src/or/proto_socks.c

```diff
@@ -477,8 +600,8 @@ handle_socks_message(const uint8_t *raw_data, size_t datalen, socks_request_t *r
       goto end;
     }
     /* RFC1929 SOCKS5 username/password subnegotiation. */
-    if ((!req->got_auth && raw_data[0] == 1) ||
-        req->auth_type == SOCKS_USER_PASS) {
+    if (!req->got_auth && (raw_data[0] == 1 ||
```

Comment:
I don't understand the reasoning behind this change?

Reply:
I don't remember exactly, but it has something to do with how our implicit SOCKS FSM works. If I change it back, `5_authenticate_with_data` starts failing.

This code will go away when we rework our SOCKS implementation to function like an explicit state machine.